### PR TITLE
Added pt-PT localization.

### PR DESCRIPTION
**Pull request by tiagodenoronha**
## Description
Adding localized strings for pt-PT

### CODE REVIEW

File: packages/component/src/Localization/pt-PT.js [hunk 1]
```diff
@@ -1,64 +1,86 @@
...
-    return new Intl.DateTimeFormat('pt-PT').format(date);
-  } else {
     return date.toLocaleString('pt-PT', {
-      day: '2-digit',
-      hour: '2-digit',
-      hour12: false,
-      minute: '2-digit',
-      month: '2-digit',
-      year: 'numeric',
+        day: '2-digit',
```
Comment (corinagum) [inline packages/component/src/Localization/pt-PT.js]: remove extra spacing

File: packages/component/src/Localization/pt-PT.js [hunk 2]
```diff
@@ -1,64 +1,86 @@
...
+        hour12: false,
+        minute: '2-digit',
+        month: '2-digit',
+        year: 'numeric',
     });
-  }
+}
+
+function botSaidSomething(avatarInitials, text, timestamp) {
+    return `Bot ${avatarInitials} said, ${text}, ${xMinutesAgo(timestamp)}`;
```
Comment (corinagum) [inline packages/component/src/Localization/pt-PT.js]: I think these need to be updated to Portuguese

File: packages/component/src/Localization/pt-PT.js [hunk 3]
```diff
@@ -1,64 +1,86 @@
...
+}
+
+function userSaidSomething(avatarInitials, text, timestamp) {
+    return `User ${avatarInitials} said, ${text}, ${xMinutesAgo(timestamp)}`;
```
Comment (corinagum) [inline packages/component/src/Localization/pt-PT.js]: these also need to be updated

**Comment by tiagodenoronha** [comment]
Yep, let me check out what he did and get him on board with the changes aswell!

**Comment by tiagodenoronha** [comment]
Merged the changes onto the PR. 
Left the "Retry" as it was because it stands for "Try again" and not "Repeat"

Yet, I found this behaviour when trying to figure out when the message was sent, shouldn't that be localized? (Image for reference)

imagem

@bodyzatva  can you give it a look and check if it's ok? :)

File: packages/component/src/Localization/pt-PT.js [hunk 4]
```diff
@@ -16,49 +18,69 @@ function xMinutesAgo(dateStr) {
...
+  'New messages': 'Novas mensagens',
+  'Right': 'Direita',
+  'retry': 'tentar novamente',
+  'Retry': 'Tentar novamente',
   'Send': 'Enviar',
-  'Sending': 'a enviar',
+  'Sending': 'A enviar',
   'Speak': 'Falar',
-  'Starting…': 'A iniciar…',
+  'Starting…': 'A iniciar...',
```
Comment (corinagum) [inline packages/component/src/Localization/pt-PT.js]: Could you switch this back to '…' (single character)

**Comment by corinagum** [comment]
That should definitely be localized. Let me look into this while we're waiting for confirmation from @bodyzatva. 
@tiagodenoronha if you have the bandwidth to help with the investigation, your help would be very appreciated! Thank you

**Comment by tiagodenoronha** [comment]
Sure, I can help no problem! Question is, how can I help? JS isn't exactly my top language to code in :)

**Comment by corinagum** [comment]
Oh no worries - we need to investigate why {retry} is showing instead of the localization, but if you aren't able to then I will get to it eventually -- hopefully by the end of next week :)

**Comment by tiagodenoronha** [comment]
Sorry! When it comes to js, I'm more of a "bother" than a "helper" :)

**Comment by corinagum** [comment]
No problem. Could you let me know what reproduction steps you're using to get the Send failed message? That will help a lot!

**Comment by tiagodenoronha** [comment]
That one is easy! Spin up bot framework emulator, and send a message with pt-BR locale. If I'm not mistaken, the Bot Framework Emulator uses Webchat behind the scenes, so if it happens there, it happens here :)

**Comment by corinagum** [comment]
Ah in that case, I think we expect it to fail since pt-BR is not available on Emulator's version of Web Chat. I'll investigate a little more soon and get back to this PR. Thanks for the help :)

Did you intend on updating both pt-PT and pt-BR?

**Comment by tiagodenoronha** [comment]
I didn't, only pt-PT, did I update both by mistake?
I admit i'm still a little green on these "Request Changes" parts...

**Comment by corinagum** [comment]
Hey @tiagodenoronha, sorry for the confusion. Let's try to clear things up before proceeding. :)

- You made changes to pr-PR (thank you!)
- You reported a bug that instead of using the string for 'retry', Web Chat displays {retry}, but based on your comments this is happening in pt-BR, not pt-PT. To clarify, is this problem occurring on pt-PT as well? Or just BR?
- if just pt-BR, I believe we could do a quick fix by changing `'Retry': '{retry}'` to `'Retry': 'Repetir'`. What do you think?

**Comment by tiagodenoronha** [comment]
- Made changes to pt-PT, yes :b
- Well, since the emulator is only available in pt-BR, I don't actually know if it happens in pt-PT, but I believe I saw this happening in en-US as well... Might need to check and confirm.
- Sound OK! And for pt-PT also!

**Comment by bodyzatva** [comment]
@tiagodenoronha  this is your final changes?  i need this changes for my project :)

**Comment by corinagum** [comment]
@bodyzatva and @tiagodenoronha, sorry for the delay! I think I figured out the issue (it was a separate bug) and will have the fix up hopefully this week. I will keep you guys updated if anything changes. Thanks for your patience!

File: packages/component/src/Localization/pt-PT.js (hunk 4 above)
Comment (corinagum) [reply packages/component/src/Localization/pt-PT.js]: marked as unresolved; the ellipses is still three characters.

Comment (tiagodenoronha) [reply packages/component/src/Localization/pt-PT.js]: Wait, I definitely need help with Github.. It shows me like I did: https://github.com/microsoft/BotFramework-WebChat/pull/2015/commits/1b62894dc0f141eb9925d5ba1c000f4f3d2d7e50

Comment (tiagodenoronha) [reply packages/component/src/Localization/pt-PT.js]: Hmm, and the branch on my repo appears good as well... https://github.com/tiagodenoronha/BotFramework-WebChat/blob/patch-1/packages/component/src/Localization/pt-PT.js

**Comment by tiagodenoronha** [comment]
Hey @corinagum , did you check my comments? :)

File: packages/component/src/Localization/pt-PT.js (hunk 4 above)
Comment (corinagum) [reply packages/component/src/Localization/pt-PT.js]: Okay that's odd. 

So I have some bad news, haha. There's going to be a PR this week that will cause merge conflicts with this PR. Do you mind if we put this on hold for a couple days, and I can alert you when the new PR is in? That way you can branch off of the new master and incorporate your changes, and hopefully this odd situation will resolve itself in the new PR. Does that sound ok? Sorry for the trouble!

**Comment by corinagum** [comment]
@tiagodenoronha, sorry for the trouble but the new merge is in. Would you mind opening a new PR with your changes? Hopefully that will resolve the non-updating commits as well. 
Closing this PR, but waiting for your next one! Thanks for the help :)

**Comment by tiagodenoronha** [comment]
Hey no problem! Give me a couple of days and I'll send you a new PR! :)

**Pull request closed without merging**